prompting/forward.py

```diff
@@ -21,16 +21,50 @@
 import numpy as np
 import bittensor as bt
 
-from typing import List
+from typing import List, Dict, Awaitable
 from prompting.agent import HumanAgent
 from prompting.dendrite import DendriteResponseEvent
 from prompting.conversation import create_task
-from prompting.protocol import PromptingSynapse
+from prompting.protocol import StreamPromptingSynapse
 from prompting.rewards import RewardResult
 from prompting.utils.uids import get_random_uids
 from prompting.utils.logging import log_event
 
 
+async def handle_response(responses: Dict[int, Awaitable]) -> List[bt.Synapse]:
+    """handles calling the async Awaitable generators.
+    Args:
+        responses (Dict[int, List[Awaitable]]): uid : Awaitable pairings.
+    Returns:
+        List[bt.Synapse]: The last chunk yielded from the bt.dendrite.call() is the original synapse
+        with the compeltion attached. Return a list of completed StreamPromptingSynapses.
+    """
+    synapses = []
+
+    for uid, resp in responses.items():
+        async for chunk in resp:
+            bt.logging.debug(f"\nchunk for uid {uid}: {chunk}")
+
+        synapse = (
+            chunk  # last object yielded is the synapse itself with completion filled
+        )
+
+        # Double check to make sure chunk is a synapse
+        if isinstance(synapse, StreamPromptingSynapse):
+            synapses.append(synapse)
+        else:
+            synapses.append(
+                StreamPromptingSynapse(
+                    roles=["user"], messages=["failure"], completion=""
+                )
+            )
+            bt.logging.debug(
+                f"Synapse is not StreamingPromptingSynapse. Miner uid {uid} completion set to '' "
+            )
+
+    return synapses
```

Comment:
I am decently confident that there are occurring async, because if you were to look at the total elapsed time to compute a response, there should be no completions of N > 1 miners queried, with a total time of > 10s that have a status code of 200. 
![image](https://github.com/opentensor/prompting/assets/83707037/ef602836-14f1-45a2-94cd-6ad2a378b820)

This data is from the validator running on testnet 102 with 2 miners. I am seeing that the total time in some individual uids be larger than 10s, and that shouldn't be happening. so maybe there is a bug here